### DISCUSSION TITLE
Update thumbnails of videos on disk (after using tagger)

### DISCUSSION THREAD

**Discussion by lamyergeier** [stashapp/stash#5457] [Q&A]
When we use tagger it downloads the relevant thumbnail for the videos from server, and displays it in stash. 

Can we set this as the thumbnail for the video on the disk?

**Comment by feederbox826** [comment]
no, it's not supported as setting the cover via XMP changes the contents of the files, breaking oshash and md5

**Reply by lamyergeier** [reply]
But that should not be a problem as the file path is not changing and a hash can be recalculated.

Additionally i don't think that perceptual hash used by stash uses thumbnail data.

**Reply by stg-annon** [reply]
>  i don't think that perceptual hash used by stash uses thumbnail data.

indeed which is why they said oshash and md5, not to mention any other 3rd party hashes people might use.

one of the core features of stash is that it does not modify your source files, If you absolutely wanted this feature it could be scripted from a plugin I doubt it would get adopted to main

**Reply by lamyergeier** [reply]
Thanks it would be nice, if atleast data can be easily exported by stash, so users can run ffmpeg commands to do it manually, but I am not sure if there exists such option. 

For example, if stash could export thumbnail as `Video_filename.jpg`, then we could easily do it by ourselves. Also, would be nice to export the video info as a `Video_filename.json` file.

> I like stash and I hope that it is not going away, I just was hoping to future proof my data. :)

**Reply by stg-annon** [reply]
the API makes this data available you would be looking for [this](https://github.com/stashapp/stash/blob/602f95dd29cb89d59c54d3917c4b97ea2d8115bc/graphql/schema/types/scene.graphql#L13)

so you could make a plugin to pull the thumbnail and add it to the file with ffmpeg you would just also need to tell stash to rescan the file after you modify it.
